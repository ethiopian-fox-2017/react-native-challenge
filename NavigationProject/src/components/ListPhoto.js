import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  Text,
  View,
  ScrollView,
  ListView,
  Image
} from 'react-native';

import { Col, Grid } from 'native-base';

import PhotoItem from './PhotoItem'
import { fetchPhoto } from '../actions'

class ListPhoto extends Component {
  constructor(props) {
    super(props)
  }

  static navigationOptions = { title : '700px'}

  componentDidMount() {
    this.props.fetchPhoto()
  }

  render () {
    return (
      <View style={styles.container}>
      <ScrollView style={styles.content}>
      <Grid>
        <Col style={{alignItems:'center'}}>
          <Image
              style={{width:80, height: 80, borderRadius: 100, margin: 10}}
              source={{uri:'https://avatars0.githubusercontent.com/u/25143119?v=3&u=23b4af0333afa56e2575ed1b72ea469fa733108c&s=400'}}
          />
        </Col>
        <Col style={{alignItems:'center', marginTop: 20}}>
          <Text style={styles.stats}>1234</Text>
          <Text style={{marginLeft:10}}>posts</Text>
        </Col>
        <Col style={{alignItems:'center', marginTop: 20}}>
          <Text style={styles.stats}>999</Text>
          <Text style={{marginLeft:10}}>followers</Text>
        </Col>
        <Col style={{alignItems:'center',  marginTop: 20}}>
          <Text style={styles.stats}>888</Text>
          <Text style={{marginLeft:10}}>following</Text>
        </Col>
      </Grid>
      <View>
        <Text style={{marginLeft: 10, color:'black', fontWeight:'bold'}}>Laksono Suryadi</Text>
        <Text style={{marginLeft: 10, color:'black', marginBottom:4}}>A photography enthusiast</Text>
        <Text style={{marginLeft: 10, marginBottom: 20, fontSize:12}}>Followed by<Text style={{fontWeight:'bold', color:'black'}}> Anthony Juan, Andra Satria, + 10 more</Text></Text>
      </View>
        {this.props.photos.length === 0 && <Text style={styles.loading}>LOADING</Text>}
        <View style={styles.grid}>
            {
              this.props.photos.map(photo => (
                <PhotoItem key={photo.id} photo={photo} navigation={this.props.navigation} />
                )
              )
            }
        </View>
      </ScrollView>
      </View>
    )
  }
}


const styles = {
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF'
  },
  content: {
    flex: 1,
    width: '100%'
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    marginTop: 20,
    marginBottom: 20
  },
  grid: {
    flexDirection: 'row',
    flexWrap:'wrap'
  },
  loading: {
    marginTop: 100,
    fontSize: 30,
    justifyContent: 'center',
    alignSelf: 'center'
  },
  stats: {
    marginLeft: 10,
    fontSize: 20,
    fontWeight: 'bold',
    color: 'black'
  }
}


const mapStateToProps = (state) => ({
  photos: state.photos
})

const mapDispatchToProps = dispatch => ({
  fetchPhoto: () => dispatch(fetchPhoto()),
})

export default connect(mapStateToProps, mapDispatchToProps)(ListPhoto);
