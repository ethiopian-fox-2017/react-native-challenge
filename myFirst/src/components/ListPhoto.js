import React, { Component } from 'react';
import {
  Image,
} from 'react-native';
import { Card, CardItem,Text, Button, Icon, Left, Body } from 'native-base';


class ListPhoto extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    const { photo } = this.props;
    return (
      <Card>
        <CardItem>
          <Left>
            <Image
              style={{width: 60, height: 60, borderRadius: 100}}
              source={{uri: photo.user.userpic_url }}
            />
            <Body>
              <Text>{photo.user.fullname}</Text>
              <Icon active name="camera" />
              <Text>{photo.camera}</Text>
            </Body>
          </Left>
        </CardItem>
        <CardItem cardBody>
          <Image
            style={{width: 360, height: 360}}
            source={{uri: photo.image_url }}
          />
        </CardItem>
        <CardItem content>
          <Text>
            {photo.description}
          </Text>
        </CardItem>
        <CardItem style={{justifyContent: 'space-around'}}>
          <Button transparent>
              <Icon active name="eye" />
              <Text>{photo.times_viewed}</Text>
          </Button>
          <Button transparent>
              <Icon active name="heart" />
              <Text>{photo.votes_count}</Text>
          </Button>
        </CardItem>

      </Card>
    )
  }
}

export default ListPhoto;
