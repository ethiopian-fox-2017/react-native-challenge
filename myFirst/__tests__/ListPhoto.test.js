import 'react-native';
import React from 'react';
import ListPhoto from '../src/components/ListPhoto';
import { shallow } from 'enzyme';
// Note: test renderer must be required after react-native.


test('renders correctly', () => {
  const data = {
    user: '',
    userpic_url: '',
  }
  const tree = shallow(
      <ListPhoto photo={data} />
  );

  expect(tree).toMatchSnapshot();
});
