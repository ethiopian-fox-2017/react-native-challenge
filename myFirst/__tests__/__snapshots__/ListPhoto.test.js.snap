// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Styled(Card)
    virtual={undefined}
>
    <Styled(CardItem)
        virtual={undefined}
    >
        <Styled(Left)
            virtual={undefined}
        >
            <Image
                source={
                    Object {
                        "uri": undefined,
                      }
                }
                style={
                    Object {
                        "borderRadius": 100,
                        "height": 60,
                        "width": 60,
                      }
                }
            />
            <Styled(Body)
                virtual={undefined}
            >
                <Styled(Text)
                    virtual={undefined}
                />
                <Styled(Icon)
                    active={true}
                    name="camera"
                    virtual={undefined}
                />
                <Styled(Text)
                    virtual={undefined}
                />
            </Styled(Body)>
        </Styled(Left)>
    </Styled(CardItem)>
    <Styled(CardItem)
        cardBody={true}
        virtual={undefined}
    >
        <Image
            source={
                Object {
                    "uri": undefined,
                  }
            }
            style={
                Object {
                    "height": 360,
                    "width": 360,
                  }
            }
        />
    </Styled(CardItem)>
    <Styled(CardItem)
        content={true}
        virtual={undefined}
    >
        <Styled(Text)
            virtual={undefined}
        />
    </Styled(CardItem)>
    <Styled(CardItem)
        style={
            Object {
                "justifyContent": "space-around",
              }
        }
        virtual={undefined}
    >
        <Styled(Button)
            transparent={true}
            virtual={undefined}
        >
            <Styled(Icon)
                active={true}
                name="eye"
                virtual={undefined}
            />
            <Styled(Text)
                virtual={undefined}
            />
        </Styled(Button)>
        <Styled(Button)
            transparent={true}
            virtual={undefined}
        >
            <Styled(Icon)
                active={true}
                name="heart"
                virtual={undefined}
            />
            <Styled(Text)
                virtual={undefined}
            />
        </Styled(Button)>
    </Styled(CardItem)>
</Styled(Card)>,
  "nodes": Array [
    <Styled(Card)
      virtual={undefined}
>
      <Styled(CardItem)
            virtual={undefined}
      >
            <Styled(Left)
                  virtual={undefined}
            >
                  <Image
                        source={
                              Object {
                                    "uri": undefined,
                                  }
                        }
                        style={
                              Object {
                                    "borderRadius": 100,
                                    "height": 60,
                                    "width": 60,
                                  }
                        }
                  />
                  <Styled(Body)
                        virtual={undefined}
                  >
                        <Styled(Text)
                              virtual={undefined}
                        />
                        <Styled(Icon)
                              active={true}
                              name="camera"
                              virtual={undefined}
                        />
                        <Styled(Text)
                              virtual={undefined}
                        />
                  </Styled(Body)>
            </Styled(Left)>
      </Styled(CardItem)>
      <Styled(CardItem)
            cardBody={true}
            virtual={undefined}
      >
            <Image
                  source={
                        Object {
                              "uri": undefined,
                            }
                  }
                  style={
                        Object {
                              "height": 360,
                              "width": 360,
                            }
                  }
            />
      </Styled(CardItem)>
      <Styled(CardItem)
            content={true}
            virtual={undefined}
      >
            <Styled(Text)
                  virtual={undefined}
            />
      </Styled(CardItem)>
      <Styled(CardItem)
            style={
                  Object {
                        "justifyContent": "space-around",
                      }
            }
            virtual={undefined}
      >
            <Styled(Button)
                  transparent={true}
                  virtual={undefined}
            >
                  <Styled(Icon)
                        active={true}
                        name="eye"
                        virtual={undefined}
                  />
                  <Styled(Text)
                        virtual={undefined}
                  />
            </Styled(Button)>
            <Styled(Button)
                  transparent={true}
                  virtual={undefined}
            >
                  <Styled(Icon)
                        active={true}
                        name="heart"
                        virtual={undefined}
                  />
                  <Styled(Text)
                        virtual={undefined}
                  />
            </Styled(Button)>
      </Styled(CardItem)>
</Styled(Card)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ListPhoto
        photo={
                Object {
                        "user": "",
                        "userpic_url": "",
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ListPhoto {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "photo": Object {
            "user": "",
            "userpic_url": "",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Styled(Card)
          virtual={undefined}
>
          <Styled(CardItem)
                    virtual={undefined}
          >
                    <Styled(Left)
                              virtual={undefined}
                    >
                              <Image
                                        source={
                                                  Object {
                                                            "uri": undefined,
                                                          }
                                        }
                                        style={
                                                  Object {
                                                            "borderRadius": 100,
                                                            "height": 60,
                                                            "width": 60,
                                                          }
                                        }
                              />
                              <Styled(Body)
                                        virtual={undefined}
                              >
                                        <Styled(Text)
                                                  virtual={undefined}
                                        />
                                        <Styled(Icon)
                                                  active={true}
                                                  name="camera"
                                                  virtual={undefined}
                                        />
                                        <Styled(Text)
                                                  virtual={undefined}
                                        />
                              </Styled(Body)>
                    </Styled(Left)>
          </Styled(CardItem)>
          <Styled(CardItem)
                    cardBody={true}
                    virtual={undefined}
          >
                    <Image
                              source={
                                        Object {
                                                  "uri": undefined,
                                                }
                              }
                              style={
                                        Object {
                                                  "height": 360,
                                                  "width": 360,
                                                }
                              }
                    />
          </Styled(CardItem)>
          <Styled(CardItem)
                    content={true}
                    virtual={undefined}
          >
                    <Styled(Text)
                              virtual={undefined}
                    />
          </Styled(CardItem)>
          <Styled(CardItem)
                    style={
                              Object {
                                        "justifyContent": "space-around",
                                      }
                    }
                    virtual={undefined}
          >
                    <Styled(Button)
                              transparent={true}
                              virtual={undefined}
                    >
                              <Styled(Icon)
                                        active={true}
                                        name="eye"
                                        virtual={undefined}
                              />
                              <Styled(Text)
                                        virtual={undefined}
                              />
                    </Styled(Button)>
                    <Styled(Button)
                              transparent={true}
                              virtual={undefined}
                    >
                              <Styled(Icon)
                                        active={true}
                                        name="heart"
                                        virtual={undefined}
                              />
                              <Styled(Text)
                                        virtual={undefined}
                              />
                    </Styled(Button)>
          </Styled(CardItem)>
</Styled(Card)>,
        "_debugID": 2,
        "_renderedOutput": <Styled(Card)
          virtual={undefined}
>
          <Styled(CardItem)
                    virtual={undefined}
          >
                    <Styled(Left)
                              virtual={undefined}
                    >
                              <Image
                                        source={
                                                  Object {
                                                            "uri": undefined,
                                                          }
                                        }
                                        style={
                                                  Object {
                                                            "borderRadius": 100,
                                                            "height": 60,
                                                            "width": 60,
                                                          }
                                        }
                              />
                              <Styled(Body)
                                        virtual={undefined}
                              >
                                        <Styled(Text)
                                                  virtual={undefined}
                                        />
                                        <Styled(Icon)
                                                  active={true}
                                                  name="camera"
                                                  virtual={undefined}
                                        />
                                        <Styled(Text)
                                                  virtual={undefined}
                                        />
                              </Styled(Body)>
                    </Styled(Left)>
          </Styled(CardItem)>
          <Styled(CardItem)
                    cardBody={true}
                    virtual={undefined}
          >
                    <Image
                              source={
                                        Object {
                                                  "uri": undefined,
                                                }
                              }
                              style={
                                        Object {
                                                  "height": 360,
                                                  "width": 360,
                                                }
                              }
                    />
          </Styled(CardItem)>
          <Styled(CardItem)
                    content={true}
                    virtual={undefined}
          >
                    <Styled(Text)
                              virtual={undefined}
                    />
          </Styled(CardItem)>
          <Styled(CardItem)
                    style={
                              Object {
                                        "justifyContent": "space-around",
                                      }
                    }
                    virtual={undefined}
          >
                    <Styled(Button)
                              transparent={true}
                              virtual={undefined}
                    >
                              <Styled(Icon)
                                        active={true}
                                        name="eye"
                                        virtual={undefined}
                              />
                              <Styled(Text)
                                        virtual={undefined}
                              />
                    </Styled(Button)>
                    <Styled(Button)
                              transparent={true}
                              virtual={undefined}
                    >
                              <Styled(Icon)
                                        active={true}
                                        name="heart"
                                        virtual={undefined}
                              />
                              <Styled(Text)
                                        virtual={undefined}
                              />
                    </Styled(Button)>
          </Styled(CardItem)>
</Styled(Card)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ListPhoto
    photo={
        Object {
            "user": "",
            "userpic_url": "",
          }
    }
/>,
}
`;
